### PR TITLE
[thread-cert] refactor case 5.2.6 using pktverify

### PR DESCRIPTION
**Pull request by FiveDimensions**

### CODE REVIEW

File: tests/scripts/thread-cert/Cert_5_2_06_RouterDowngrade.py [hunk 1]
```diff
@@ -33,175 +33,221 @@
...
         23: {
+            'name': 'ROUTER_22',
             'mode': 'rdn',
             'panid': 0xface,
             'router_downgrade_threshold': 32,
             'router_selection_jitter': 1,
             'router_upgrade_threshold': 32
         },
         ROUTER24: {
+            'name': 'ROUTER_23',
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_5_2_06_RouterDowngrade.py:250]: it's a little confusing why ROUTER24 is `ROUTER_23`. Maybe change `ROUTER24` to `ROUTER23` ?

File: tests/scripts/thread-cert/Cert_5_2_06_RouterDowngrade.py [hunk 2]
```diff
@@ -221,36 +267,107 @@ def test(self):
...
+                  ).\
+            must_next()
+        _pkt.must_not_verify(lambda p: (ADDRESS_REGISTRATION_TLV) in p.mle.tlv.type)
+
+        _pkt = pkts.filter_wpan_src64(ROUTER_1).\
+            filter_ipv6_dst(LEADER_RLOC).\
+            filter_coap_request(ADDR_REL_URI).\
+            filter(lambda p: {
+                              NL_MAC_EXTENDED_ADDRESS_TLV,
+                              NL_RLOC16_TLV
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_5_2_06_RouterDowngrade.py:349]: nit: Is it worth to verify that the RLOC16 TLV equals to Router1's RLOC16 ?

File: tests/scripts/thread-cert/Cert_5_2_06_RouterDowngrade.py [hunk 3]
```diff
@@ -221,36 +267,107 @@ def test(self):
...
+                   ).\
+           must_next()
+
+        # Step 4: Leader receives Address Release messages and sends a2.04
+        #         Changed CoAP response.
+
+        pkts.filter_wpan_src64(LEADER).\
+            filter_ipv6_dst(ROUTER_1_RLOC).\
+            filter_coap_ack(ADDR_REL_URI).\
+            filter(lambda p: p.coap.code == COAP_CODE_ACK).\
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_5_2_06_RouterDowngrade.py:360]:
```suggestion
```
`filter_coap_ack` already verifies that the COAP code is `ACK`.

File: tests/scripts/thread-cert/Cert_5_2_06_RouterDowngrade.py (hunk 2 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_5_2_06_RouterDowngrade.py:351]: added the resign rloc16 check

File: tests/scripts/thread-cert/Cert_5_2_06_RouterDowngrade.py (hunk 1 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_5_2_06_RouterDowngrade.py]: changed to ROUTER23

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**